### ISSUE TITLE
TargetRevision field is patched even if the source type is set to Helm

### ISSUE THREAD

**Issue by aokomorowski** [dag-andersen/argocd-diff-preview#33]
Hi Dag, I found this one while working on #29, I couldn't figure out why would I receive this error:

```
❌ name-of-the-deployment, Failed to load target state: failed to generate manifest for source 2 of 3: rpc error: code = Unknown desc = invalid revision 'main': improper constraint: main
```

I found the exact place [where this error is thrown](https://github.com/argoproj/argo-cd/blob/master/reposerver/repository/repository.go#L2386) in the ArgoCD repository and I wondered why would be a chart revision set to the main (and why does ArgoCD use Helm Chart flow here). Finally, I found that ArgoCD Diff Preview is patching all targetRevision fields in Applications/ApplicationSets with the branch name.

Let's say we've got an ApplicationSet with multiple sources defined:
```
apiVersion: argoproj.io/v1alpha1
kind: ApplicationSet
metadata:
  name: application
  namespace: argocd
spec:
  [ OMITTED ]
  template:
    spec:
      sources:
        - repoURL: ghcr.io/repository
          chart: chart-name
          targetRevision: 0.1.6 <-- gets patched as well
        - repoURL: git@github.com:org/repo.git
          targetRevision: main
          ref: values

```

ArgoCD-Diff-Preview will patch both `targetRevision`s under `.spec.template.spec.sources[*]`, which might not be expected. I'd recommend dealing with it like folks from ArgoCD did - by [checking](https://github.com/argoproj/argo-cd/blob/master/pkg/apis/application/v1alpha1/types.go#L271) if `chart` field is not empty.

I'll be happy to provide more details if needed, I appreciate your work on this project!

**Comment by dag-andersen** [comment]
Hi @aokomorowski,
Thank you for reporting this!

I actually thought the `chart` check would be unnecessary, but I can definitely add it 🚀 

[Current Code](https://github.com/dag-andersen/argocd-diff-preview/blob/832871d2e8f3d2f5d8f0e2a98872f20adde46b1c/src/parsing.rs#L109-L121): For each source under `.spec.sources` it will check if the `repoURL` contains the name of the repository, and if so, it will patch that `targetRevision`. That it how it knows if the source should be patched or not. 

So, I'm just curious: Does your `repoURL` (in your helm chart source) contain the name of your `org/repo`? 😄 If not, then I am very confused as to why it would patch your helm chart entry.

**Comment by aokomorowski** [comment]
It does, GHCR registries come in the format of ghcr.io/organisation/name-of-repository 😬

**Comment by dag-andersen** [comment]
Nice catch! - I'll fix it later today or tomorrow :)

**Comment by dag-andersen** [comment]
@aokomorowski I believe this is now fixed in [v0.0.12](https://github.com/dag-andersen/argocd-diff-preview/releases/tag/v0.0.12) 🚀 Let me know if this didn’t resolve the issue :)

**Comment by aokomorowski** [comment]
Hell yeah, that worked! Thanks, I've finally managed to get the diffs & the manifests out of it. I think we can close both issues
